include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -1466,87 +1477,91 @@
       if (!t) {
         // Spin waiting for work.
         for (int i = 0; i < spin_count && !t && !done_; i++) {
-          if (((i+1)%steal_count == 0)) {
+          if (((i + 1) % steal_count == 0)) {
             t = Steal(StealAttemptKind::TRY_ONE);
           } else {
             t = q.PopFront();
           }
+          if (spin_loop_status_.load(std::memory_order_relaxed) == kIdle) {
```

Comment:
>spin_loop_status_ [](http://example.com/codeflow?start=14&length=17)

Do we have to create a new static member for the purpose?
Could "allow_spinning_" be bring into "while" loop to indicate whether to continue the running spinning. And StopSpin() and StopSpin() operate on "allow_spinning_" for controlling the behavior.

Reply:
allow_spinning_ has different meaning, it is a constant property of this TP instance taken fromconfiguration. spin_loop_status_ is a runtime instant control to stop/ or allow restarting spinning when Run() calls commence/stop. We want to control inner loop and not outer loop.
By default, when the feature is not enabled, it should behave as usual.